Example/Tests/Tests.swift

```diff
@@ -1,27 +1,114 @@
+        let testRealm = try! Realm(configuration: Realm.Configuration(inMemoryIdentifier: "Test"))
+        repository = RealmRepository<QuickEmployee>(realm: testRealm)
     }
     
     override func tearDown() {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        let testRealm = try! Realm(configuration: Realm.Configuration(inMemoryIdentifier: "Test"))
+        repository = RealmRepository<QuickEmployee>(realm: testRealm)
         super.tearDown()
     }
     
-    func testExample() {
-        // This is an example of a functional test case.
+    func testGetAll() {
+        addRandomeMockObjects(repo: repository)
+
+        repository?.getAll({ (allObjects) in
+            XCTAssert(allObjects.count == 4)
+        })
     }
-    
-    func testPerformanceExample() {
-        // This is an example of a performance test case.
-        self.measure() {
-            // Put the code you want to measure the time of here.
-        }
+
+
+    func testDeleteAll() {
+        addRandomeMockObjects(repo: repository)
+
+        var deletionError: Error?
+        repository?.deleteAll({ error in
+            deletionError = error
+        })
+
+        repository?.getAll({ allObjects in
+            XCTAssert(allObjects.isEmpty && deletionError == nil)
+        })
+    }
+
+    func testFilter(){
+
+        repository?.create(QuickEmployee(name: "Torsten", age: 19, data: Data()), { _ in })
+        repository?.create(QuickEmployee(name: "Torben", age: 21, data: Data()), { _ in })
+        repository?.create(QuickEmployee(name: "Tim", age: 87, data: Data()), { _ in })
+        repository?.create(QuickEmployee(name: "Struppi", age: 3, data: Data()), { _ in })
+
+        let newEmployeeName = "Zementha"
+        repository?.create(QuickEmployee(name: newEmployeeName, age: 34, data: Data()), { _ in })
+
+        repository?.getElements(filteredBy: "name = %@", newEmployeeName, completion: { (filteredEmployees) in
+            guard let firstEmployee = filteredEmployees.first else { return }
+
+            let correctEmployee = firstEmployee.name == newEmployeeName
+            let containsOneEmployee = filteredEmployees.count == 1
+
+            XCTAssert(correctEmployee && containsOneEmployee)
+        })
+    }
+
+
+    func testSorting(){
+
+        let tim = QuickEmployee(name: "Tim", age: 87, data: Data())
+        repository?.create(tim, { _ in })
+
+        let struppi = QuickEmployee(name: "Struppi", age: 3, data: Data())
+        repository?.create(struppi, { _ in })
+
+        let torsten = QuickEmployee(name: "Torsten", age: 19, data: Data())
+        repository?.create(torsten, { _ in })
+
+        let torben = QuickEmployee(name: "Torben", age: 21, data: Data())
+        repository?.create(torben, { _ in })
+
+        repository?.getElements(sorted: \QuickEmployee.age, ascending: false, completion: { (filteredEmployees) in
```

Comment:
I think you should test ascending: true also 🤔

Reply:
ok